lucene/core/src/java/org/apache/lucene/codecs/lucene80/IndexedDISI.java

```diff
@@ -0,0 +1,601 @@
+      buffer.clear(0, buffer.length());
+      prevBlock++;
+    }
+    final int lastBlock = prevBlock == -1 ? 0 : prevBlock; // There will always be at least 1 block (NO_MORE_DOCS)
+    // Last entry is a SPARSE with blockIndex == 32767 and the single entry 65535, which becomes the docID NO_MORE_DOCS
+    // To avoid creating 65K jump-table entries, only a single entry is created pointing to the offset of the
+    // NO_MORE_DOCS block, but with the jumpBlockIndex set to the logical EMPTY block after all real blocks.
+    jumps = addJumps(jumps, out.getFilePointer()-origo, DocIdSetIterator.NO_MORE_DOCS & 0xFFFF0000,
+        lastBlock, lastBlock+1);
+    buffer.set(DocIdSetIterator.NO_MORE_DOCS & 0xFFFF);
+    flush(DocIdSetIterator.NO_MORE_DOCS >>> 16, buffer, 1, denseRankPower, out);
+    // offset+index jump-table stored at the end
+    return flushBlockJumps(jumps, lastBlock+1, out, origo);
+  }
+
+  // Adds entries to the offset & index jump-table for blocks
+  private static int[] addJumps(int[] jumps, long offset, int index, int startBlock, int endBlock) {
+    assert offset < Integer.MAX_VALUE : "Logically the offset should not exceed 2^30 but was >= Integer.MAX_VALUE";
+    jumps = ArrayUtil.grow(jumps, (endBlock+1)*2);
+    for (int b = startBlock; b < endBlock; b++) {
+      jumps[b*2] = index;
+      jumps[b*2+1] = (int) offset;
+    }
+    return jumps;
+  }
+
+  // Flushes the offet & index jump-table for blocks. This should be the last data written to out
+  // This method returns the blockCount for the blocks reachable for the jump_table or -1 for no jump-table
+  private static short flushBlockJumps(int[] jumps, int blockCount, IndexOutput out, long origo) throws IOException {
+    if (blockCount == 2) { // Jumps with a single real entry + NO_MORE_DOCS is just wasted space so we ignore that
+      blockCount = 0;
+    }
+    for (int i = 0 ; i < blockCount ; i++) {
+      out.writeInt(jumps[i*2]); // index
+      out.writeInt(jumps[i*2+1]); // offset
+    }
+    // As there are at most 32k blocks, the count is a short
+    // The jumpTableOffset will be at lastPos - (blockCount * Long.BYTES)
+    return (short)blockCount;
+  }
+
+  /** The slice that stores the {@link DocIdSetIterator}. */
+  private final IndexInput slice;
+  private final int jumpTableEntryCount;
+  private final byte denseRankPower;
+  private final RandomAccessInput jumpTable; // Skip blocks of 64K bits
+  private final byte[] denseRankTable;
+  private final long cost;
+
+  /**
+   * @param in backing data.
+   * @param offset starting offset for blocks in backing data.
+   * @param length the number of bytes in the backing data.
+   * @param jumpTableEntryCount the number of blocks convered by the jump-table.
+   * @param cost normally the number of logical docIDs.
+   */
+  IndexedDISI(IndexInput in, long offset, long length, int jumpTableEntryCount, byte denseRankPower, long cost) throws IOException {
+    this(in.slice("docs", offset, length),
```

Comment:
it _seems_ to include the jump table in the end, can we make this slice only cover blocks, but not the jump table?